qiskit_nature/operators/second_quantization/qubit_converter.py

```diff
@@ -40,58 +106,235 @@ def z2symmetry_reduction(self) -> Optional[List[int]]:
 
-        if self.z2symmetry_reduction is None:
+        did_2_q = False
+        if (self._two_qubit_reduction and
+                self._mapper.allows_two_qubit_reduction and
+                num_particles is not None):
+            two_q_reducer = TwoQubitReduction(num_particles)
+            reduced_qubit_ops = [cast(PauliSumOp, two_q_reducer.convert(op)) for op in qubit_ops]
+            did_2_q = True
+        else:
             reduced_qubit_ops = qubit_ops
-            self._z2symmetries = Z2Symmetries([], [], [], None)
+
+        return reduced_qubit_ops, did_2_q
+
+    def _find_symmetry_reduce(self, qubit_ops: List[PauliSumOp])\
+            -> Tuple[List[Optional[PauliSumOp]], Z2Symmetries]:
+
+        if self.z2symmetry_reduction is None:
+            tapered_qubit_ops = qubit_ops
+            z2_symmetries = self._no_symmetries
+        else:
+            z2_symmetries = Z2Symmetries.find_Z2_symmetries(qubit_ops[0])
+            if z2_symmetries.is_empty():
+                logger.debug('No Z2 symmetries found')
+                tapered_qubit_ops = qubit_ops
+            else:
+                # check sector definition fits to symmetries found
+                if len(self._z2symmetry_reduction) != len(z2_symmetries.symmetries):
+                    raise QiskitNatureError('z2symmetry_reduction tapering values list has '
+                                            'invalid length {} should be {}'.
+                                            format(len(self._z2symmetry_reduction),
+                                                   len(z2_symmetries.symmetries)))
+                # Check all operators commute with main operator's symmetry
+                logger.debug('Checking operators commute with symmetry:')
+                symmetry_ops = []
+                for symmetry in z2_symmetries.symmetries:
+                    symmetry_ops.append(PauliSumOp.from_list([(symmetry.to_label(), 1.0)]))
+                commuted = []
+                for i, qubit_op in enumerate(qubit_ops):
+                    commutes = QubitConverter._check_commutes(symmetry_ops, qubit_op)
+                    if i == 0 and not commutes:
+                        raise QiskitNatureError('Z2 symmetry failure. Main operator must commute '
+                                                'with symmetries found from it!')
+                    commuted.append(commutes)
+                    logger.debug("Qubit operators commuted with symmetry %s", commuted)
+
+                # set tapering for the given sector and taper operators
+                z2_symmetries.tapering_values = self._z2symmetry_reduction
+                tapered_qubit_ops = []
+                for i, commutes in enumerate(commuted):
+                    op = z2_symmetries.taper(qubit_ops[i]) if commutes else None
+                    tapered_qubit_ops.append(op)
+
+        return tapered_qubit_ops, z2_symmetries
+
+    def _symmetry_reduce(self, qubit_ops: List[PauliSumOp])\
```

Comment:
This entire function is almost an exact duplicate of the previous one.
If my comment about merging `convert` and `convert_more` will not be pursued these two functions can probably be split more conveniently in any case in order to minimize duplication.

Reply:
I do not know. They are different but similar - one needs to find symmetries the other not, etc. Yes one could do it with set of if elses. It was easier to do it like this for now since they are separate

Reply:
Yes, let's postpone internal refactorings until after the upcoming release